controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -155,125 +150,140 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 			}
 
-			// Unmarshal the response datum as externalResource.
-			for _, resource := range resources {
+			// Go over the received objects and operate on them accordingly.
+			for _, resourceItem := range resources {
 				data := map[string]string{}
-				err = json.Unmarshal(resource.Data, &data)
+				err = json.Unmarshal(resourceItem.Data, &data)
 				if err != nil {
 					return reconcile.Result{}, fmt.Errorf("failed to unmarshal StorageClassClaim configuration response: %v", err)
 				}
-				externalResources = append(externalResources, ExternalResource{
-					Name: resource.Name,
+				resource := storagecluster.ExternalResource{
+					Name: resourceItem.Name,
 					Data: data,
-					Kind: resource.Kind,
-				})
-			}
-
-			// Go over the received objects and operate on them accordingly.
-			var enableRookCSICephFS bool
-			var availableStorageClassConfigurations []StorageClassConfiguration
-			var storageClassClaimOwnerReference metav1.OwnerReference
-			for _, resource := range externalResources {
-
-				// storageClassClaimOwnerReference points to the StorageClassClaim object as the owner object of the created resource(s).
-				storageClassClaimOwnerReference = metav1.OwnerReference{
-					APIVersion: r.storageClassClaim.APIVersion,
-					Kind:       r.storageClassClaim.Kind,
-					Name:       r.storageClassClaim.Name,
-					UID:        r.storageClassClaim.UID,
+					Kind: resourceItem.Kind,
 				}
+
 				switch resource.Kind {
 				case "Secret":
-					objectMeta := metav1.ObjectMeta{
-						Name:      resource.Name,
-						Namespace: r.storageClassClaim.Namespace,
-						OwnerReferences: []metav1.OwnerReference{
-							storageClassClaimOwnerReference,
-						},
-					}
-					secret := &corev1.Secret{
-						ObjectMeta: objectMeta,
-						Data:       make(map[string][]byte),
-					}
+					secret := &corev1.Secret{}
+					secret.Name = resource.Name
+					secret.Namespace = r.storageClassClaim.Namespace
+					secret.Data = map[string][]byte{}
 					for key, value := range resource.Data {
 						secret.Data[key] = []byte(value)
 					}
-					secretTemplate := &corev1.Secret{ObjectMeta: objectMeta}
-					err := r.get(secretTemplate)
+					err := r.own(secret)
 					if err != nil {
-						if errors.IsNotFound(err) {
-							r.log.Info("Creating a new Secret", "Secret", secret.Name)
-							if err := r.Client.Create(r.ctx, secret); err != nil {
-								return reconcile.Result{}, fmt.Errorf("failed to create Secret: %v", err)
-							}
-						} else {
-							return reconcile.Result{}, fmt.Errorf("failed to get Secret: %v", err)
+						return reconcile.Result{}, fmt.Errorf("failed to own Secret: %v", err)
+					}
+					_, err = controllerutil.CreateOrUpdate(r.ctx, r.Client, secret, func() error {
+						return nil
+					})
+					if err != nil {
+						return reconcile.Result{}, fmt.Errorf("failed to create or update secret %v: %s", secret, err)
+					}
+				case "StorageClass":
```

Comment:
This is something I'd like a few clarifications upon. As per my understanding around https://github.com/red-hat-storage/ocs-operator/pull/1608#discussion_r845805399, I've removed SCConfig and everything related to it. Was this more or less the desired change?